enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,297 @@
+1. The administrator will perform the debug report creation procedure:
+```bash
+sudo sos report
+```
+
+For the CI, the same procedure applies.
+
+#### Variation [optional]
+N/A
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+N/A
+
+### Risks and Mitigations
+There are four main risks:
+1. Upgrades and incompatible changes.
+1. `sos` release cadence.
+1. Extra required disk space.
+1. CEE having a different tool.
+
+#### Upgrades
+When upgrading a MicroShift system we may need to upgrade just MicroShift, or
+the OS as well. If it is the latter then `sos` will be updated before
+MicroShift, therefore we must ensure backwards compatibility with the plugin.
+
+If it is the former then we need to ensure newer MicroShift versions do not
+introduce incompatible changes with the reporting capabilities of the `sos`
+plugin. This may also be seen from a different perspective: the plugin should
+make no assumptions about names or locations of components whenever possible.
+In other words, the plugin must be written in the most generic way that still
+allows retrieval of relevant information.
+
+Still, if incompatible changes are introduced then we need to upgrade `sos` as
+well.
+
+#### Sos release cadence
+`sos` release cadence has been shortened to 4 weeks starting on Feb 1st 2023.
+This change has been introduced to allow faster support of new components and
+features. Since MicroShift follows the same release calendar as OpenShift,
+`sos` is faster.
+
+New versions of `sos` will simply slot in to the next regular release. If there
+is an urgent need a hotfix can be made between releases.
+
+#### Extra disk space
+Adding a plugin to `sos` and not relying on MicroShift binary means it needs
+the `oc` or `kubectl` binary installed. This package is available in the RHEL
+repos, but it comes at an extra cost of 125MB once installed.
+
+If we compare it against the minimum disk size required (which is 8GB at the
+time of this writing), it represents an extra 1.5%.
+
+The greenboot feature for MicroShift 4.13 requires installing `oc` as well, so
+the first enhancement to be implemented will introduce it for the other one.
+
+#### CEE having a different tool
+CEE might have concerns over using a different tool than `must-gather`. `sos`
+reports are able to extract a full system report, as opposed to `must-gather`
+which is only suitable for OpenShift clusters.
+
+Moreover, there is a growing number of plugins to support anything that runs
+on a RHEL system, making it a suitable go-to gathering tool when it comes to
+diagnosing system-wide issues.
+
+### Drawbacks
+Discussed in risks and mitigations.
+
+## Design Details
+Refer to `sos` [documentation](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin)
+for how to create plugins.
+
+The file structure should ressemble that of must-gather.
+
+### Open Questions [optional]
+1. Should `/var/lib/microshift` contents be added to the report? They contain
```

Comment:
If this directory is used for the [CNI/CSI enhancement](https://github.com/openshift/enhancements/pull/1304) manifests then it would be future proof, provided we skip any sensitive data file.

Reply:
How will the plugin know what data is sensitive?

Reply:
Its up to what you put in the plugin. For example, openshift plugin skips all certificates and keys.